File: week2_code/linear/LinearSearch.java
package linear;
import java.util.Scanner;

public class LinearSearch {

    public static int linearSearch(int[] listOfNumber,int sizeOfNumberList,int elementToSearch){
        for(int index=0;index<sizeOfNumberList;index++){
            if(listOfNumber[index] == elementToSearch){
                return index;
            }
        }
        return -1;
    }

    public static void main(String[] args) {
        int[] listOfNumber = {5,34,23,44,55,32,65,69,65,32,67,94,23};

        Scanner keyboard = new Scanner(System.in);

        while(true){
            System.out.print("Enter a number you want to search (press -1 to quit): ");
            int elementToSearch = keyboard.nextInt();

            if(elementToSearch == -1){
                break;
            }

            if(linearSearch(listOfNumber,listOfNumber.length,elementToSearch) == -1){
                System.out.println("element is not present");
            }else{
                System.out.println("element present");
            }
        }


    }
}

File: week4/linear searxh/linearSearch.py
def linear_Search(listOfNumbers, lengthOfList, key):  
  
    for i in range(0, lengthOfList):  
        if (listOfNumbers[i] == key):  
            return i  
    return -1  
  
  
listOfNumbers = [12,23,34,45,56,67,78,89,91]  
key = input("Enter a key to search : ") 
key = int(key)
  
lengthOfList = len(listOfNumbers)  
res = linear_Search(listOfNumbers, lengthOfList, key)  
if(res == -1):  
    print("Element not found")  
else:  
    print("Element found at index: ", res)  